apps/backend/src/auth/StfcUserAuthorization.ts

```
@@ -164,46 +172,67 @@ export class StfcUserAuthorization extends UserAuthorization {
-        logger.logWarn(rethrowMessage, {
-          cause: error,
-          token: token,
-        });
-
-        throw new GraphQLError(rethrowMessage);
+    const loginBySession: LoginDTO =
+      await UOWSClient.sessions.getLoginBySessionId(token);
+    if (!loginBySession) {
+      const rethrowMessage =
+        'Failed to fetch user details for STFC external authentication';
+      logger.logWarn(rethrowMessage, {
+        token: token,
       });
+    }
+    const stfcUserTemp: BasicPersonDetailsDTO[] | null =
+      await UOWSClient.basicPersonDetails
+        .getBasicPersonDetails(undefined, undefined, undefined, [
+          loginBySession.userId,
+        ])
+        .then((rawStfcUser) => rawStfcUser)
+        .catch((error) => {
+          const rethrowMessage =
+            'Failed to fetch user details for STFC external authentication';
+          logger.logWarn(rethrowMessage, {
+            cause: error,
+            token: token,
+          });
 
-    if (!stfcUser) {
+          throw new GraphQLError(rethrowMessage);
+        });
+
+    if (!stfcUserTemp) {
       logger.logInfo('No user found for STFC external authentication', {
         externalToken: token,
       });
 
       return null;
     }
 
+    const stfcUser: StfcBasicPersonDetails | null = toStfcBasicPersonDetails(
+      stfcUserTemp[0]
+    );
+
     // Create dummy user if one does not exist in the proposals DB.
     // This is needed to satisfy the FOREIGN_KEY constraints
     // in tables that link to a user (such as proposals)
-    const userNumber = parseInt(stfcUser.userNumber);
-    const dummyUser =
-      await this.userDataSource.ensureDummyUserExists(userNumber);
+    let userNumber;
+    let dummyUser;
+    if (stfcUser!.userNumber != null) {
+      userNumber = parseInt(stfcUser!.userNumber);
+      dummyUser = await this.userDataSource.ensureDummyUserExists(userNumber);
+    } else {
+      throw error;

```

Can we also throw this error with a message .